### PR TITLE
Document Rust support

### PR DESCRIPTION
**Pull request by chriskuehl**

### CODE REVIEW

File: index.mako [hunk 1]
```diff
@@ -471,6 +472,28 @@ produce an executable that will match the `entry` – usually through
 __Support:__ ruby hooks work without any system-level dependencies.  It has
 been tested on linux and macOS and _may_ work under cygwin.
 
+### rust
+
+_new in **TBD**_
```
Comment (chriskuehl) [inline index.mako]: reminder to update this before merging

Comment (asottile) [reply index.mako]: it'll be 1.10.0 :)

File: index.mako [hunk 2]
```diff
@@ -471,6 +472,28 @@ produce an executable that will match the `entry` – usually through
...
+pre-commit cache, not polluting your user-level Cargo installations).
+
+When specifying `additional_dependencies` for Rust, you can use the syntax
+`{package_name}:{package_version}` to specify a new library dependency (used to
+build _your_ hook repo), or the special syntax
+`cli:{package_name}:{package_version}` for a CLI dependency (built separately,
+with binaries made available for use by hooks).
+
+__Support:__ Rust hooks currently require a pre-existing Rust installation.  It
+has been tested on linux and Windows, and probably works under macOS.
```
Comment (asottile) [inline index.mako]:
```
asottile@Macbook-Pro:/tmp/pre-commit$ PATH="$HOME/.cargo/bin:$PATH" tox -e py36 -- tests -k rust
GLOB sdist-make: /private/tmp/pre-commit/setup.py
py36 create: /private/tmp/pre-commit/.tox/py36
py36 installdeps: -rrequirements-dev.txt
py36 inst: /private/tmp/pre-commit/.tox/dist/pre_commit-1.9.0.zip
py36 installed: aspy.yaml==1.1.1,atomicwrites==1.1.5,attrs==18.1.0,cached-property==1.4.2,cfgv==1.0.0,coverage==4.5.1,flake8==3.5.0,identify==1.0.18,mccabe==0.6.1,mock==2.0.0,more-itertools==4.2.0,nodeenv==1.3.0,pbr==4.0.3,pluggy==0.6.0,pre-commit==1.9.0,py==1.5.3,pycodestyle==2.3.1,pyflakes==1.6.0,pytest==3.6.0,pytest-env==0.6.2,PyYAML==3.12,six==1.11.0,toml==0.9.4,virtualenv==16.0.0
py36 runtests: PYTHONHASHSEED='2981025304'
py36 runtests: commands[0] | coverage erase
py36 runtests: commands[1] | coverage run -m pytest tests -k rust
============================= test session starts ==============================
platform darwin -- Python 3.6.5, pytest-3.6.0, py-1.5.3, pluggy-0.6.0
rootdir: /private/tmp/pre-commit, inifile: tox.ini
plugins: env-0.6.2
collected 500 items / 490 deselected                                           

tests/repository_test.py .....                                           [ 50%]
tests/languages/all_test.py .....                                        [100%]

================== 10 passed, 490 deselected in 72.82 seconds ==================
```

File: index.mako (hunk 1 above)
Comment (chriskuehl) [reply index.mako]: updated, thanks!

File: index.mako (hunk 2 above)
Comment (chriskuehl) [reply index.mako]: \o/

**Pull request merged**